### DISCUSSION TITLE
How to use gitlens.remotes in settings.json for github repo

### DISCUSSION THREAD

**Discussion by guochen-code** [gitkraken/vscode-gitlens#3282] [Q&A]
I have this setup in the settings.json file in the VS code, but have not seen any effect. Wondering how to use gitlens.remotes with any public GitHub repo (in this example: integrate with vscode-gitlens repo on github). What else should I configure?

{
    "gitlens.integrations.enabled": true,

    "gitlens.remotes": [
        {
            "domain": "github.com",
            "type": "GitHub",
            "protocol": "https",
            "name": "GitHub",
            "urls": {
                "repository": "https://github.com/gitkraken/vscode-gitlens",
                "branches": "https://github.com/gitkraken/vscode-gitlens/branches",
                "branch": "https://github.com/gitkraken/vscode-gitlens/tree/{branch}",
                "commit": "https://github.com/gitkraken/vscode-gitlens/commit/{sha}",
                "file": "https://github.com/gitkraken/vscode-gitlens/blob/{branch}/{file}",
                "fileInBranch": "https://github.com/gitkraken/vscode-gitlens/blob/{branch}/{file}",
                "fileInCommit": "https://github.com/gitkraken/vscode-gitlens/blob/{sha}/{file}",
                "fileLine": "https://github.com/gitkraken/vscode-gitlens/blob/{branch}/{file}#L{line}",
                "fileRange": "https://github.com/gitkraken/vscode-gitlens/blob/{branch}/{file}#L{start}-L{end}"

                
            }
        }
    ],
}

**Comment by funwhilelost** [comment]
Looking pretty close @guochen-code - you can get some hints on the syntax in VS Code by hovering over the value string.  It shows you also need to use `{repo}` so your example would be:

Also `{sha}` should be `{id}`

```json
"gitlens.remotes": [
    {
        "domain": "github.com",
        "type": "GitHub",
        "protocol": "https",
        "name": "GitHub",
        "urls": {
            "repository": "https://github.com/gitkraken/{repo}",
            "branches": "https://github.com/gitkraken/{repo}/branches",
            "branch": "https://github.com/gitkraken/{repo}/tree/{branch}",
            "commit": "https://github.com/gitkraken/{repo}/commit/{id}",
            "file": "https://github.com/gitkraken/{repo}/blob/{branch}/{file}",
            "fileInBranch": "https://github.com/gitkraken/{repo}/blob/{branch}/{file}",
            "fileInCommit": "https://github.com/gitkraken/{repo}/blob/{sha}/{file}",
            "fileLine": "https://github.com/gitkraken/{repo}/blob/{branch}/{file}#L{line}",
            "fileRange": "https://github.com/gitkraken/{repo}/blob/{branch}/{file}#L{start}-L{end}",   
        }
    }
],
```

**Comment by h0uter** [comment]
For me it also wasnt working. I figured out I had to set the "type" to "Custom", instead of "GitLab".
